src/Cli.Tests/ConfigureOptionsTests.cs

```diff
@@ -0,0 +1,123 @@
+
+            ILoggerFactory loggerFactory = TestLoggerSupport.ProvisionLoggerFactory();
+
+            SetLoggerForCliConfigGenerator(loggerFactory.CreateLogger<ConfigGenerator>());
+            SetCliUtilsLogger(loggerFactory.CreateLogger<Utils>());
+        }
+
+        /// <summary>
+        /// Validates that `dab configure` is a no-op because the user provided no options to update
+        /// the config file. The config file must not change.
+        /// </summary>
+        [TestMethod]
+        public void TestNoUpdateRuntimeSettings()
+        {
+            // Arrange
+            _fileSystem!.AddFile(TEST_RUNTIME_CONFIG_FILE, new MockFileData(INITIAL_CONFIG));
+
+            // Act: Run Configure with no options
+            ConfigureOptions options = new(
+                depthLimit: null,
+                config: TEST_RUNTIME_CONFIG_FILE
+            );
+
+            Assert.IsTrue(_fileSystem!.File.Exists(TEST_RUNTIME_CONFIG_FILE));
+            Assert.IsTrue(TryConfigureSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            string updatedConfig = _fileSystem!.File.ReadAllText(TEST_RUNTIME_CONFIG_FILE);
+
+            // Assert that INITIAL_CONFIG is same as the updated config
+            Assert.IsTrue(JToken.DeepEquals(JObject.Parse(INITIAL_CONFIG), JObject.Parse(updatedConfig)));
+        }
+
+        /// <summary>
+        /// Tests that running "dab configure --runtime.graphql.depth-limit 8" on a config with no depth-limit previously specified, results
+        /// in runtime.graphql.depthlimit property being set with the value 8.
+        [TestMethod]
+        public void TestAddDepthLimitForGraphQL()
+        {
+            // Arrange
+            _fileSystem!.AddFile(TEST_RUNTIME_CONFIG_FILE, new MockFileData(INITIAL_CONFIG));
+            int maxDepthLimit = 8;
+
+            Assert.IsTrue(_fileSystem!.File.Exists(TEST_RUNTIME_CONFIG_FILE));
+
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config));
+            Assert.IsNull(config.Runtime!.GraphQL!.DepthLimit);
+
+            // Act: Attmepts to Add Depth Limit
+            ConfigureOptions options = new(
+                depthLimit: maxDepthLimit,
+                config: TEST_RUNTIME_CONFIG_FILE
+            );
+            Assert.IsTrue(TryConfigureSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert: Validate the Depth Limit is added
+            string updatedConfig = _fileSystem!.File.ReadAllText(TEST_RUNTIME_CONFIG_FILE);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config.Runtime!.GraphQL!.DepthLimit);
+            Assert.AreEqual(maxDepthLimit, config!.Runtime!.GraphQL!.DepthLimit);
+        }
+
+        /// <summary>
+        /// Test to update the current depth limit for GraphQL and removal the depth limit using -1.
```

Comment:
nit: more detail about validation.

---
When runtime.graphql.depth-limit has an initial value of 8, validates that 
"dab configure --runtime.graphql.depth-limit {value}" sets the expected depth limit.
---